### PR TITLE
Refactor content script, fix scroll issues, and update dependencies

### PR DESCRIPTION
**Pull request by blakkd**
*Disclaimer: this was ~mainly~ - actually entirely - vibe coded using opencode/Qwen3.6-27B. I just spent some time for decision making and testing along the way.*

*And thanks for all your prior work, can't do without this extension!*

## Summary
A set of improvements to the extension — the website (`sites/`) was not touched.
### Content Script: Svelte → Vanilla JS
Replaced the Svelte-based content script with a plain JS class (`AutoScroll.js`). This simplifies the build pipeline, removes the Svelte compiler dependency for the content script, and reduces bundle size.
### Scroll Fixes
- Added support for `flex-col-reverse` layouts (previously scrolled in the wrong direction)
- Added support for `overflow: visible` containers
- Fixed scroll range calculation for more accurate scrolling
### Vulnerability Reduction (28 → 3)
Replaced `svelte-intl` with a lightweight custom i18n module and removed `pngquant` — together these two packages accounted for nearly all critical and high vulnerabilities, plus most moderate ones.
### Dependency Updates
Updated all dependencies to their latest versions, resolving remaining critical and moderate security issues across the board.
### Code Consolidation
Merged `math.js`, `cursor.js`, and `event.js` into a single `utils.js` — they were small, tightly coupled helpers that didn't warrant separate files.
### Build Fix
Added `caniuse-lite` to resolve a browserslist warning during pnpm builds.

### CODE REVIEW

**Comment by blakkd** [comment]
The main motivation was that recent changes to llama.cpp webUI broke the scrolling behavior. Then I saw https://github.com/Aurailus/ClickScroll was still working I thought we could look how to handle it here too. I preferred continuing from your work because it was already more robust overall.

**Comment by ubermanu** [comment]
Hey,

Overall good rework, my only concern would be that the overlay is not running in a shadomDOM anymore (to avoid style clashes with the main document)

What you can do is either transform your AutoScroll class into a WebComponent, or create a completely new WC for the overlay (that would appear on mousedown?)

**Comment by blakkd** [comment]
Hey @ubermanu ! Honestly I have a really limited understanding of what I'm doing... I could of course try to dirty incorporate your suggestion relying on my agent again, but as it sounds, I think we should avoid that :D
So I guess I won't be able to pursue further.
Actually I was a bit hesitant to make this PR because it was several commits in a row and felt like slope PR... But thinking again I finally did it because it solved my issue and llama.cpp is quite popular (though I don't think many ppl use the webUI, despite it being better and better over time), and finally also because it wasn't pure slope and took me a few hours...
So, as I respect your previsou work, I think you should treat this PR as a take it or leave it. Also, I saw your repo wasn't touched for some times before, and I don't want to overwhelm you with this, having to dive back in an partially forgotten codebase :/


Do what you think is best, I won't take anything personally!

**Comment by blakkd** [comment]
Oh and plus: I realize I did no test at all on Firefox!

**Comment by blakkd** [comment]
I finally went back in... 2 new commits:

*Qwen3.6-27B to you:*
1. Overlay as a Web Component [9a02ff8](https://github.com/ubermanu/roller/pull/20/commits/9a02ff8a45b0e9f848ac761bd21cfc0ff126cb1a) — created a dedicated `<roller-overlay>` custom element with Shadow DOM, so the overlay's static styles (position, z-index, etc.) are now fully encapsulated and can't be overridden by host page CSS.
2. Content script compatibility fix [baaefdd](https://github.com/ubermanu/roller/pull/20/commits/baaefddfefe300abb6142361e3901d10b1147cee) — turns out `customElements` is null in Chromium's content script isolated world, so the WC couldn't be registered. Added a fallback that creates a plain `<div>` with an identical property interface (`bgImage`, `bgPosition`, `cursor`) when the WC can't be used. Firefox uses the proper WC with Shadow DOM, Chromium falls back to the div — both work correctly.

Also fixed a `chrome.storage.local` null access issue that was causing crashes on some pages.

And Firefox tested this time.

**Comment by blakkd** [comment]
I just remembered Firefox has this `autoscrolling` option in Settings -> General. So I'm wondering if is still relevant to support Firefox:

image

**Comment by blakkd** [comment]
Sorry again but I couldn't let the codebase in the state it was when I saw the option menu was broken, and then saw the scroll on links was too!
So 3 new commits:

- Hotfix: *(Option menu not visible)* Migrated the options page to Svelte 5.

- Fix: *(Middle-click scrolling was not working on links despite `scrollOnLinks: true`)* The issue was caused by the `<a>` element's `.host` property (which returns the hostname string for hyperlinks) being incorrectly treated as a shadow DOM root, breaking scroll target detection.

- Feat *(opinionated)*: `scrollOnLinks: true` by default.

PS: actually I the extension still has a purpose for Firefox users: precisely this scroll on links feature!

**Comment by blakkd** [comment]
I'm sorry to add so much on top of the initial PR... I just understood the purpose and practical benefit of git branches instead of doing it all on `main`!
I have been moving fast and just wanted to tackle as many edge case as possible and was in a mental tunnel mode...
Now that deep, I don't know if it could even be realistic to checkout and create new branches for so many commits... I don't know what to do to properly contribute now.
Again, really I don't want to put any hassle on you, so if you don't have the time or energy to look into all this, or are concerned about AI slop, I'd totally understand.
I keep the PR open for now, but feel free to close it if that's too much! I feel like a kid now coming back to mommy and mommy has to figure out :D

**Comment by blakkd** [comment]
In all case, here is what I can observe on my side in the current state: no new edge case found since yesterday during my daily browsing, apart one:
- embedded youtube video, when initiating scrolling (middleclick/CTRL-click) on it, we are for some reason sent to the top of the current page.
 
I coudn't find why. I gave up on it for now.
Other than that, not an edge case, but the reset button in the extension menu UI isn't triggering the actual reset of the values, but that's minor and should of course be solved in one go, just didn't focus on it so far... I didn't test with your last release, but might have broke it when transitioning away from svelte

**Comment by blakkd** [comment]
done, also set those as default values:

image

**Comment by ubermanu** [comment]
Hey @blakkd don't stress yourself too much, it's ok :smile: 
I don't mind you pushing LLM generated code as long as you take the time to understand and test it (which you do)

I'll try to make the web component work this WE (couldn't find 1min to do so during the week)

File: extensions/roller/src/contentScript.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
-import AutoScroll from './components/AutoScroll'
+import '@webcomponents/custom-elements'
```
Comment (ubermanu) [inline extensions/roller/src/contentScript.ts:1]: @blakkd this was the missing piece for it to work

**Comment by ubermanu** [comment]
@blakkd merged it as is, don't hesitate to open a new pr/issue if you find some bugs :smiley:

**Comment by blakkd** [comment]
> I don't mind you pushing LLM generated code as long as you take the time to understand and test it (which you do)

Honestly I don't at all! Yes I do test, but other than that, I am in a thick mist! I usually only touch python where I slowly start to partially understand what I'm doing, but web dev is abstract art for me!

If I ever find new failing cases, I'll report first as an issue, and eventually try to fix keeping PR small and scoped.

**Comment by blakkd** [comment]
And thanks a lot for having looked into my mess!
